### PR TITLE
Added getter for children of CompoundIngredient

### PR DESCRIPTION
**Pull request by mcenderdragon**
This PR adds a getter to CompoundIngredient to access the collection of sub ingredients. ~~I am not sure if we should make this collection readonly or not.~~ The collection is readonly because otherwise someone could break cashing.

### CODE REVIEW

**Comment by Wyn-Price** [comment]
`getChilds` should be `getChildren`, as children is the proper plural for child. Having it as a Collection is fine, the children list should be unmodified anyway

**Comment by mcenderdragon** [comment]
@Wyn-Price doen. I kind of though about JS were it was named childs.

File: src/main/java/net/minecraftforge/common/crafting/CompoundIngredient.java [hunk 1]
```diff
@@ -109,4 +109,10 @@ public boolean isSimple()
     {
         return isSimple;
     }
+    
+    @Nonnull
+    public Collection<Ingredient> getChildren()
+    {
+        return this.children;
```
Comment (mezz) [inline src/main/java/net/minecraftforge/common/crafting/CompoundIngredient.java]: use `return Collections.unmodifiableCollection(this.children);`

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by mcenderdragon** [comment]
ok @mezz done.

**Comment by mezz** [comment]
Looks good. What do you need this for?

**Review by mezz — APPROVED** [review]

**Comment by mcenderdragon** [comment]
I am building a small mod for recipe improvment, so it will log all recipe were items have an oredict but the oredict version is not used. But unfortunetly it is not possible to say if a recipe uses oredict or not if the compound ingredient is used (the other classes are Ingredient and OreIngredient). The goal is to make a config so for example large modpacks can just list some recipes and I will replace them so they are oredict compatible for a better experience. (Nothing is worse then broken OreDict mod support)

**Comment by LexManos** [comment]
I dont think this is needed, as it breaks the Ingredient api, but whatever if I dont do this people'll just reflect and break crap anyways.

**Pull request merged**